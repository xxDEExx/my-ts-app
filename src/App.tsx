import React, { SFC, Fragment, useEffect } from 'react';
import { ThemeProvider } from '@material-ui/styles';
import CssBaseline from '@material-ui/core/CssBaseline';

import config from 'config';

import theme, { useStyles } from 'themes';
import Route from 'routes';

const App: SFC = () => {
	useStyles();

	useEffect(() => {
		document.title = config.title;
	});

	return (
		<Fragment>
			<CssBaseline />
			<ThemeProvider theme={theme}>
				<Route />
			</ThemeProvider>
		</Fragment>
	);
}

export default App;
