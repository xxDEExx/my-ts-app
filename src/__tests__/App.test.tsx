import React from "react";
import { render, cleanup } from '@testing-library/react';
import { ConnectedRouter } from 'connected-react-router';
import { createBrowserHistory } from 'history';

import ReduxProvider from 'store';
import LocalizeProvider from 'config/locale/Provider';

const history = createBrowserHistory();

import App from 'App';

const renderComponent = () => {
    return render(
        <ReduxProvider>
            <LocalizeProvider>
                <ConnectedRouter history={history}>
                    <App />
                </ConnectedRouter>
            </LocalizeProvider>
        </ReduxProvider>
    )
}

describe('APP', () => {
    afterEach(cleanup);

    it("render with no issue", () => {
        const { asFragment } = renderComponent();
        expect(asFragment()).toMatchSnapshot();
    })
});
