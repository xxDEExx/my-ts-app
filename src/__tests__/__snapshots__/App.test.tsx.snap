// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP render with no issue 1`] = `
<DocumentFragment>
  <h1
    class="makeStyles-header-1"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
      />
    </svg>
    <span>
      Dashboard Page.
    </span>
  </h1>
  <span>
    Loaded
  </span>
  <br />
  <span>
    <b>
      0
    </b>
  </span>
  <br />
  <div
    aria-label="outlined primary button group"
    class="MuiButtonGroup-root"
    role="group"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButtonGroup-grouped MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedPrimary MuiButton-outlined MuiButton-outlinedPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Increment
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButtonGroup-grouped MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedPrimary MuiButton-outlined MuiButton-outlinedPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Decrement
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
  <br />
  <br />
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
        />
      </svg>
       Move to about
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
